### PR TITLE
Add metrics integration to istio charms

### PR DESCRIPTION
**Pull request by IbraAoad**
## Issue
Istio exposes metrics for each workload it deploys:

- **`istiod`**: Accessible at `istiod.{self.model.name}.svc:15014`, on port `15014` at `/metrics`.
- **`ztunnel`**: Exposes metrics on port `15020` at `/metrics`, with no FQDN available.
- **`istio-cni-node`**: Exposes metrics on port `15014` at `/metrics`, with no FQDN available.

## Solution
This PR adds metrics integration through the [prometheus_scrape](https://charmhub.io/prometheus-k8s/libraries/prometheus_scrape) interface, utilizing the newly created [metrics-k8s-proxy](https://github.com/canonical/metrics-k8s-proxy) to aggregate these metrics into Prometheus via a unified endpoint.

## Testing Instructions

1. juju deploy istio-k8s --trust
2. juju deploy prometheus-k8s --trust
3. juju relate istio-k8s prometheus-k8s
4. Explore Prometheus to validate that the metrics are present. Additionally, check for the `up` metric for each Istio workload.

### CODE REVIEW

**Comment by IbraAoad** [comment]
Implementation is ready for review

**TODO**: Fix proxy image location + integration tests once https://github.com/canonical/metrics-k8s-proxy/pull/1 and https://github.com/canonical/metrics-proxy-rock/pull/1 are merged

File: src/charm.py [hunk 1]
```diff
@@ -62,6 +64,8 @@
 GATEWAY_API_CRDS_LABEL = "gateway-apis-crds"
 GATEWAY_API_CRDS_RESOURCE_TYPES = {CustomResourceDefinition}
 
+METRICS_LABELS = {"charms.canonical.com/istio.io.metrics": "aggregated"}
```
Comment (dstathis) [inline src/charm.py:67]: Do we want to include the app name and model uuid here? While it's unlikely we will support two instances of istio on the same mesh, making the label unique to this app could leave that option open for the future.

File: src/charm.py [hunk 2]
```diff
@@ -74,19 +78,56 @@ def __init__(self, *args):
...
         self._lightkube_field_manager: str = self.app.name
+        # Configure Observability
+        self._scraping = MetricsEndpointProvider(
+            self,
+            relation_name="metrics-endpoint",
+            jobs=[{"static_configs": [{"targets": ["*:15090"]}]}],
+        )
 
         self.framework.observe(self.on.config_changed, self._reconcile)
         self.framework.observe(self.on.remove, self._remove)
```
Comment (dstathis) [inline src/charm.py:93]: We should observe pebble_ready. It's unlikely to matter but if all of our initial hooks run before pebble is ready, we will never set up the layer.

File: src/charm.py [hunk 3]
```diff
@@ -242,6 +286,22 @@ def _modify_istio_cni_configmap(self, resources: List[AnyResource]) -> str:
         # Convert the modified resources back to a YAML string
         return resources  # pyright: ignore
 
+    def _modify_pod_labels(self, resources: List[AnyResource]) -> List[AnyResource]:
```
Comment (dstathis) [inline src/charm.py:289]: Can we call this method `_add_metrics_labels` or something like that?

**Review by dstathis — APPROVED** [review]

**Pull request merged**